src/components/presentational/NavigationBar/NavigationBar.tsx

```diff
@@ -1,88 +1,123 @@
-import React, { useRef, useLayoutEffect, useContext } from 'react'
-import "./NavigationBar.css"
-import MyDataHelps from "@careevolution/mydatahelps-js"
+import React, { CSSProperties, useContext, useLayoutEffect, useRef } from 'react'
+import './NavigationBar.css'
+import MyDataHelps from '@careevolution/mydatahelps-js'
 import '@fortawesome/fontawesome-svg-core/styles.css';
 import { faChevronLeft } from '@fortawesome/free-solid-svg-icons/faChevronLeft'
-import language from "../../../helpers/language"
+import language from '../../../helpers/language'
 import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon'
 import { ColorDefinition, resolveColor } from '../../../helpers/colors';
 import { LayoutContext } from '../Layout';
 
+export type NavigationBarButtonPosition = 'left' | 'right';
+
 export interface NavigationBarProps {
 	title?: string;
 	subtitle?: string;
 	showBackButton?: boolean;
+	backButtonText?: string;
+	backButtonColor?: ColorDefinition;
+	backButtonBackgroundColor?: ColorDefinition;
 	showCloseButton?: boolean;
-	children?: React.ReactNode;
 	closeButtonText?: string;
-	backButtonText?: string;
+	closeButtonColor?: ColorDefinition;
+	closeButtonBackgroundColor?: ColorDefinition;
+	closeButtonPosition?: NavigationBarButtonPosition;
+	showSaveButton?: boolean;
+	saveButtonText?: string;
```

Comment:
This feels like a ton of additional properties here which complicate usage of this, when really it is just that you have 2 buttons you want to define right?  

Why not just have a "navigationBarLeft" and "navigationBarRight" props that are ReactNodes...and when specified they basically replace the left (back) / right (close) buttons...they would not have a handler or anything by default, you would specify that in the node.

I think that would serve advanced use cases a lot better, otherwise we will end up having to continually customize this as we come up with new button ideas...like what if we want 2 buttons on the right etc.

Reply:
That sounds good to me.  I'll make the adjustment.

I may not "replace" the existing buttons, though, since it may be a little odd to have a "showBackButton" property that doesn't actually do anything when set to `true`.

Instead, as a user of the component, when using the custom left/right nav bar nodes, I would just leave the built-in buttons hidden.

Reply:
I have made the suggested changes.  Please take another look when you have a moment.